switchbot/devices/lock.py

```diff
@@ -86,77 +83,78 @@ async def verify_encryption_key(
+            timeout=5,
+        )
+
+        if result.status_code > 299:
+            raise SwitchbotApiError(
+                f"Unexpected status code returned by SwitchBot API: {result.status_code}"
+            )
+
+        response = json.loads(result.content)
+        if response["statusCode"] != 100:
+            raise SwitchbotApiError(
+                f"{response['message']}, status code: {response['statusCode']}"
+            )
+
+        return response["body"]
+
     @staticmethod
     def retrieve_encryption_key(device_mac: str, username: str, password: str):
         """Retrieve lock key from internal SwitchBot API."""
         device_mac = device_mac.replace(":", "").replace("-", "").upper()
-        msg = bytes(username + SWITCHBOT_APP_COGNITO_POOL["AppClientId"], "utf-8")
-        secret_hash = base64.b64encode(
-            hmac.new(
-                SWITCHBOT_APP_COGNITO_POOL["AppClientSecret"].encode(),
-                msg,
-                digestmod=hashlib.sha256,
-            ).digest()
-        ).decode()
-
-        cognito_idp_client = boto3.client(
-            "cognito-idp", region_name=SWITCHBOT_APP_COGNITO_POOL["Region"]
-        )
+
         try:
-            auth_response = cognito_idp_client.initiate_auth(
-                ClientId=SWITCHBOT_APP_COGNITO_POOL["AppClientId"],
-                AuthFlow="USER_PASSWORD_AUTH",
-                AuthParameters={
-                    "USERNAME": username,
-                    "PASSWORD": password,
-                    "SECRET_HASH": secret_hash,
+            auth_result = SwitchbotLock.api_request(
+                "account",
+                "account/api/v1/user/login",
+                {
+                    "clientId": SWITCHBOT_APP_CLIENT_ID,
+                    "username": username,
+                    "password": password,
+                    "grantType": "password",
+                    "verifyCode": "",
                 },
             )
-        except cognito_idp_client.exceptions.NotAuthorizedException as err:
-            raise SwitchbotAuthenticationError(
-                f"Failed to authenticate: {err}"
-            ) from err
+            auth_headers = {"authorization": auth_result["access_token"]}
         except Exception as err:
-            raise SwitchbotAuthenticationError(
-                f"Unexpected error during authentication: {err}"
-            ) from err
+            raise SwitchbotAuthenticationError(f"Authentication failed: {err}") from err
 
-        if (
-            auth_response is None
-            or "AuthenticationResult" not in auth_response
-            or "AccessToken" not in auth_response["AuthenticationResult"]
-        ):
-            raise SwitchbotAuthenticationError("Unexpected authentication response")
+        try:
+            userinfo = SwitchbotLock.api_request(
+                "account", "account/api/v1/user/userinfo", {}, auth_headers
+            )
+            region = userinfo["botRegion"]
```

Comment:
It is recommended that you use "us" when this field is an empty string; otherwise, the domain name will not be able to make requests.